src/sdk/object_sdk.js

```diff
@@ -71,8 +72,13 @@ class ObjectSDK {
     }
 
     async _get_bucket_namespace(name) {
-        const { ns } = await bucket_namespace_cache.get_with_cache({ sdk: this, name });
-        return ns;
+        return bucket_namespace_cache.get_with_cache({ sdk: this, name });
+    }
+
+    set_namespace_bucket_path(params, bucket) {
+        if (bucket && bucket.namespace && bucket.namespace.write_resource.path) {
+            params.path = bucket.namespace.write_resource.path;
```

Comment:
I don't understand - isn't this the same as the value we are keeping in namespace_fs this.fs_path? Why do you need to set it on the params too??

Reply:
OK so I see now, the path from inside the resource vs the path that each bucket sets to use from the resource.
But both are still properties of the namespace, so we should initialize the namespace this.fs_path to already join these two together.

Reply:
fixed.